core/amber/src/main/scala/edu/uci/ics/texera/web/service/JobStatsService.scala

```diff
@@ -23,21 +26,41 @@ import edu.uci.ics.texera.web.model.websocket.event.{
 import edu.uci.ics.texera.web.storage.JobStateStore
 import edu.uci.ics.texera.web.storage.JobStateStore.updateWorkflowState
 import edu.uci.ics.texera.web.workflowruntimestate.FatalErrorType.EXECUTION_FAILURE
-import edu.uci.ics.texera.web.workflowruntimestate.{OperatorWorkerMapping, WorkflowFatalError}
+import edu.uci.ics.texera.web.workflowruntimestate.{
+  OperatorRuntimeStats,
+  OperatorWorkerMapping,
+  WorkflowFatalError
+}
 import edu.uci.ics.texera.web.workflowruntimestate.WorkflowAggregatedState.{COMPLETED, FAILED}
 
 import java.time.Instant
+import edu.uci.ics.texera.workflow.common.WorkflowContext
+import org.jooq.types.UInteger
+
+import java.util
 
 class JobStatsService(
     client: AmberClient,
-    stateStore: JobStateStore
+    stateStore: JobStateStore,
+    workflowContext: WorkflowContext
 ) extends SubscriptionManager
     with LazyLogging {
+  final private lazy val context = SqlServer.createDSLContext()
+  private val workflowRuntimeStatisticsDao = new WorkflowRuntimeStatisticsDao(context.configuration)
 
   registerCallbacks()
 
   addSubscription(
     stateStore.statsStore.registerDiffHandler((oldState, newState) => {
+      storeRuntimeStatistics(newState.operatorInfo.zip(oldState.operatorInfo).collect {
+        case ((newId, newStats), (oldId, oldStats)) =>
+          val res = OperatorRuntimeStats(
+            newStats.state,
+            newStats.inputCount - oldStats.inputCount,
```

Comment:
I see only the diffs in the input/output count are stored in each record. What's the reason behind it? Just curious.

Reply:
Personally, I thought that diffs are more useful than accumulated results for the users and users can easily sum up all the diffs if users want accumulated results.

Reply:
I see. That makes sense.